docs/user-guide/misc/visualization.md

```diff
@@ -0,0 +1,65 @@
+# Visualization
+
+Data in a Polars `DataFrame` can be visualized using common visualization libraries.
+
+<div style="display:none">
+```python exec="on" result="text" session="user-guide/misc/visualization"
+--8<-- "python/user-guide/misc/visualization.py:setup"
+```
+</div>
+
+We illustrate plotting capabilities using the Iris dataset. We scan a CSV and then do a group-by on the `species` column and get the mean of the `petal_length`.
+
+{{code_block('user-guide/misc/visualization','dataframe',[])}}
+
+```python exec="on" session="user-guide/misc/visualization"
+--8<-- "python/user-guide/misc/visualization.py:dataframe"
+```
+
+### Built-in plotting with hvPlot
+
+Polars has a `plot` method to create interactive plots using hvPlot.
+
+{{code_block('user-guide/misc/visualization','hvplot_show_plot',[])}}
+
+```python exec="on" session="user-guide/misc/visualization"
+--8<-- "python/user-guide/misc/visualization.py:hvplot_make_plot"
+```
+
+## Matplotlib
+
+To create a bar chart we can pass columns of a `DataFrame` directly to Matplotlib as a `Series` for each column. Matplotlib does not have explicit support for Polars objects but Matplotlib can accept a Polars `Series` because it can iterate through each `Series` in a similar way to a Python `list`.
```

Comment:
how about

> because it can convert each `Series` to a numpy array, which is zero-copy for numeric
> data without null values

just because it looks like they're converting to numpy array

https://github.com/matplotlib/matplotlib/blob/f0c4c41d1a08bc3f53824cddc8fef7f88fc52a43/lib/matplotlib/axes/_axes.py#L2465

and then presumably doing some C magic? I think mentioning iterating over Python lists makes it sounds like mpl is less efficient than it is :)

Reply:
Done